website/content/en/docs/best-practices/observability-best-practices.md

```diff
@@ -0,0 +1,268 @@
+
+### Operators Observability General Guidelines
+**Important:** It is highly recommended to separate your monitoring code from your core operator code.  
+We recommend to create a dedicated `/monitoring` subfolder that will include all the code of the [Operator Observability Recommended Components](#operator-observability-recommended-components), that are outlined above. For example, in the [memcached-operator](https://github.com/operator-framework/operator-sdk/tree/master/testdata/go/v4-alpha/monitoring/memcached-operator/monitoring).
+
+In your core operator code only call the functions the functions that will update the metrics value from you desired location. For example, in the [memcached-operator](https://github.com/operator-framework/operator-sdk/blob/367bd3597c30607099aa73637f5286f7120b847a/testdata/go/v3/monitoring/memcached-operator/controllers/memcached_controller.go#L242).
+
+All operators start small.This separation will help you, as a developer, with easier maintenance of both your operator core code and the monitoring code and for other stakeholders to understand your monitoring code better.
+
+### Metrics Guidelines
+
+#### Metrics Naming
+Note: The metrics naming is based on the [OpenMetrics](https://github.com/OpenObservability/OpenMetrics) and [Prometheus exposition formats](https://prometheus.io/docs/instrumenting/exposition_formats/) and adds to it.  
+
+Your operator metrics should align with the Kubernetes metrics names.
+
+Your operator users should get the same experience when searching for a metric across Kubernetes operators, resources and custom resources.
+1. Check if a similar Kubernetes metric, for node, container or pod, exists and try to align to it.
+2. The metrics search list in the Prometheus and Grafana UI is sorted in alphabetical order.
+When searching for a metric, it should be easy to identify metrics that are related to a specific operator.
+That is why we recommend that your operator metrics name to follow this format:
+`Operator name` prefix + the `Sub operator name` or `entity` + `metric name` based on the [Prometheus naming conventions](https://prometheus.io/docs/practices/naming/). For example, in the [memcached-operator](https://github.com/operator-framework/operator-sdk/blob/0d2fa86f0d3cc92c4672cb9e1d246efaefcf7ced/testdata/go/v4-alpha/monitoring/memcached-operator/monitoring/metrics.go#L14).
+
+**Note:** In Kubernetes metrics are separated like this:
+- node_network_**receive**_packets_total
+- node_network_**transmit**_packets_total  
+ 
+In this example based on `receive` and `transmit`.
+
+Please follow the same principle and don't put similar metrics details as labels, so the user experience would be fluent.
+Example for this in an operator:
+- kubevirt_vmi_network_**receive**_errors_total
+- kubevirt_vmi_network_**transmit**_bytes_total
+- kubevirt_migrate_vmi_**data_processed**_bytes
+- kubevirt_migrate_vmi_**data_remaining**_bytes
+
+#### Metrics `Help` message
+
```

Comment:
If this is talking about the `help` field defined in a certain metric, then I'm not sure if it is better to be short and efficient?
That is to say: no need to specify: `metric type`, `unique labels`.

See[ OpenMetrics Ref](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#help)

Reply:
I updated the test. I did leave the `unique labels`, since they can be important for understanding the metric, but its optional.

Reply:
I like the idea to describe `important labels`!